.github/workflows/codeql-daily.yml

```diff
@@ -21,27 +26,114 @@ jobs:
           distribution: temurin
           java-version: 17
 
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
       - name: Initialize CodeQL
         uses: github/codeql-action/init@v3
         with:
-          languages: java, c-cpp
+          languages: java
+
+      - name: Build Java code
+        # skipping build cache is needed so that all modules will be analyzed
+        run: ./gradlew assemble --no-build-cache
+
+      - name: Perform CodeQL analysis
+        uses: github/codeql-action/analyze@v3
+        with:
+          category: java
+
+  # ===== C++ Analysis Job =====
+  analyze-cpp:
+    name: "Analyze C++ Code"
+    permissions:
+      actions: read
+      security-events: write
+    runs-on: windows-latest
+    
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Set up Java 17 (required for JNI compilation)
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: 17
+
+      - name: Setup Visual Studio Build Tools
+        uses: microsoft/setup-msbuild@v1
+      
+      # This step uses Microsoft's vswhere tool to verify that the official Windows 10 SDK (version 19041) is installed.
+      # vswhere is a Microsoft-provided command-line utility that locates Visual Studio installations and their components.
+      - name: Verify Windows SDK installation
+        run: |
+          & "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe" -products * -requires Microsoft.VisualStudio.Component.Windows10SDK.19041 -property installationPath
+        shell: pwsh
 
       - name: Setup Gradle
         uses: gradle/actions/setup-gradle@v4
 
-      - name: Assemble
-        # skipping build cache is needed so that all modules will be analyzed
-        run: ./gradlew assemble --no-build-cache
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v3
+        with:
+          languages: cpp
+          debug: true
+
+      - name: Build C++ code
+        shell: cmd
+        run: |
+          "%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe" -latest -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath > vs.txt
+          set /p VSPATH=<vs.txt
+          set VCVARS=%VSPATH%\VC\Auxiliary\Build\vcvars64.bat
+          call "%VCVARS%"
+          set APPINSIGHTS_WIN10_SDK_PATH=C:\Program Files (x86)\Windows Kits\10
+          set APPINSIGHTS_VS_PATH=%VSPATH%
```

Comment:
looks unused?

```suggestion
```

Reply:
This line is needed because `APPINSIGHTS_VS_PATH` is referenced by the other files:

https://github.com/microsoft/ApplicationInsights-Java/blob/7c234b7401c1f9de863cbea8d6fe2aecd30b7951/etw/native/build.gradle#L17-L20

Reply:
I didn't see the gradle build being run below?